src/crush/CrushCompiler.cc

```diff
@@ -815,6 +815,44 @@ void CrushCompiler::dump(iter_t const& i, int ind)
     dump(i->children.begin() + j, ind+1); 
 }
 
+int CrushCompiler::adjust_bucket_item_place(iter_t const &i)
+{
+  map<string,set<string> > bucket_items;
+  map<string,iter_t> bucket_itrer;
+  vector<string> buckets;
+  for (iter_t p = i->children.begin(); p != i->children.end(); p++) {
+    if ((int)p->value.id().to_long() == crush_grammar::_bucket) {
+      string name = string_node(p->children[1]);
+      buckets.push_back(name);
+      bucket_itrer[name] = p;
+      for (unsigned q=3; q < p->children.size()-1; q++) {
+        iter_t sub = p->children.begin() + q;
+        string tag = string_node(sub->children[0]);
+        if (tag == "item") {
```

Comment:
could you compare the `id().to_long()` with `_bucket_item` also, instead of hardwiring it to "item"?